File: src/main/java/com/atsuishio/superbwarfare/compat/realcamera/RealCameraCompatHolder.java
package com.atsuishio.superbwarfare.compat.realcamera;

import com.atsuishio.superbwarfare.compat.CompatHolder;
import com.xtracr.realcamera.RealCameraCore;
import com.xtracr.realcamera.util.CrosshairUtil;
import net.fabricmc.loader.api.FabricLoader;

public class RealCameraCompatHolder {

    public static boolean hasMod() {
        return FabricLoader.getInstance().isModLoaded(CompatHolder.REALCAMERA);
    }

    public static float getCompatMoveX(float moveX) {
        if (RealCameraCore.isActive()) {
            moveX += (float) CrosshairUtil.offset.x();
        }
        return moveX;
    }

    public static float getCompatMoveY(float moveY) {
        if (RealCameraCore.isActive()) {
            moveY -= (float) CrosshairUtil.offset.y();
        }
        return moveY;
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/compat/realcamera/RealCameraCompatHolder.kt
package com.atsuishio.superbwarfare.compat.realcamera

import com.atsuishio.superbwarfare.compat.CompatHolder
import com.xtracr.realcamera.RealCameraCore
import com.xtracr.realcamera.util.CrosshairUtil
import net.fabricmc.loader.api.FabricLoader

object RealCameraCompatHolder {
    @JvmStatic
    fun hasMod(): Boolean {
        return FabricLoader.getInstance().isModLoaded(CompatHolder.REALCAMERA)
    }

    @JvmStatic
    fun getCompatMoveX(moveX: Float): Float {
        var moveX = moveX
        if (RealCameraCore.isActive()) {
            moveX += CrosshairUtil.offset.x().toFloat()
        }
        return moveX
    }

    @JvmStatic
    fun getCompatMoveY(moveY: Float): Float {
        var moveY = moveY
        if (RealCameraCore.isActive()) {
            moveY -= CrosshairUtil.offset.y().toFloat()
        }
        return moveY
    }
}